arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxFinalizationSection.java

```diff
@@ -75,39 +70,39 @@ public void resolvePostTransaction(
     final boolean coinbaseWarmth = txMetadata.isCoinbaseWarmAtTransactionEnd();
 
     final Address senderAddress = senderSnapshotBeforeFinalization.address();
-    senderSnapshotAfterTxFinalization = AccountSnapshot.canonical(hub, senderAddress);
+    senderSnapshotAfterTxFinalization = AccountSnapshot.canonical(hub, world, senderAddress);
     senderSnapshotAfterTxFinalization.turnOnWarmth(); // purely constraints based
 
     final Address recipientAddress = recipientSnapshotBeforeFinalization.address();
-    recipientSnapshotAfterTxFinalization = AccountSnapshot.canonical(hub, recipientAddress);
+    recipientSnapshotAfterTxFinalization = AccountSnapshot.canonical(hub, world, recipientAddress);
     recipientSnapshotAfterTxFinalization.turnOnWarmth(); // purely constraints based
 
     final Address coinbaseAddress = coinbaseSnapshotBeforeTxFinalization.address();
-    coinbaseSnapshotAfterFinalization = AccountSnapshot.canonical(hub, coinbaseAddress);
+    coinbaseSnapshotAfterFinalization = AccountSnapshot.canonical(hub, world, coinbaseAddress);
```

Comment:
Using the correct world to make snapshots. This world is provided to us in `traceContextExit` (in the unexceptional case.)